src/imitation/algorithms/preference_comparisons.py

```diff
@@ -377,31 +377,37 @@ def forward(
         self,
         fragment_pairs: Sequence[TrajectoryPair],
         ensemble_member_index: Optional[int] = None,
-    ):
+    ) -> Tuple[th.Tensor, Optional[th.Tensor]]:
         """Computes the preference probability of the first fragment for all pairs.
 
         Note: This function passes the gradient through for non-ensemble models.
               For an ensemble model, this function should not be used for loss
-              calculation. It can be used in case when passing the gradient is not
-              required like during active selection and inference time.
-              The EnsembleTrainer passes each member through this function instead of
-              passing the EnsembleNetwork.
+              calculation. It can be used in case where passing the gradient is not
+              required such as during active selection or inference time.
+              Therefore, the EnsembleTrainer passes each member network through this
+              function instead of passing the EnsembleNetwork object with the use of
+              `ensemble_member_index`.
 
         Args:
             fragment_pairs: batch of pair of fragments.
-            ensemble_member_index: index of member network in ensemble model
+            ensemble_member_index: index of member network in ensemble model.
+                If the model is an ensemble of networks, this cannot be None.
+
+        Raises:
+            ValueError: `uncertainty_on` not in logit|probability|label.
```

Comment:
I think this is meant to be about `ensemble_member_index` instead? (That's the only ValueError I see in this function)